Source/Extensions/Blazorise.SignaturePad/SignaturePad.razor.cs

```diff
@@ -0,0 +1,283 @@
+            var imageTypeChanged = parameters.TryGetValue<SignaturePadImageType>( nameof( ImageType ), out var paramImageType ) && ImageType != paramImageType;
+            var imageQualityChanged = parameters.TryGetValue<double?>( nameof( ImageQuality ), out var paramImageQuality ) && ImageQuality != paramImageQuality;
+            var includeImageBackgroundColorChanged = parameters.TryGetValue<bool>( nameof( IncludeImageBackgroundColor ), out var paramIncludeImageBackgroundColor ) && IncludeImageBackgroundColor != paramIncludeImageBackgroundColor;
+            var readOnlyChanged = parameters.TryGetValue<bool>( nameof( ReadOnly ), out var paramReadOnly ) && ReadOnly != paramReadOnly;
+
+            if ( dotSizeChanged
+                || minLineWidthChanged
+                || maxLineWidthChanged
+                || throttleChanged
+                || minDistanceChanged
+                || backgroundColorChanged
+                || penColorChanged
+                || velocityFilterWeightChanged
+                || imageTypeChanged
+                || imageQualityChanged
+                || includeImageBackgroundColorChanged
+                || readOnlyChanged )
+            {
+                ExecuteAfterRender( async () => await JSModule.UpdateOptions( ElementRef, ElementId, new
+                {
+                    DotSize = new { Changed = dotSizeChanged, Value = paramDotSize },
+                    MinLineWidth = new { Changed = minLineWidthChanged, Value = paramMinWidth },
+                    MaxLineWidth = new { Changed = maxLineWidthChanged, Value = paramMaxWidth },
+                    Throttle = new { Changed = throttleChanged, Value = paramThrottle },
+                    MinDistance = new { Changed = minDistanceChanged, Value = paramMinDistance },
+                    BackgroundColor = new { Changed = backgroundColorChanged, Value = paramBgColor },
+                    PenColor = new { Changed = penColorChanged, Value = paramPenColor },
+                    VelocityFilterWeight = new { Changed = velocityFilterWeightChanged, Value = paramVelocityFilterWeight },
+                    ImageType = new { Changed = imageTypeChanged, Value = ToImageTypeString( paramImageType ) },
+                    ImageQuality = new { Changed = imageQualityChanged, Value = paramImageQuality },
+                    IncludeImageBackgroundColor = new { Changed = includeImageBackgroundColorChanged, Value = paramIncludeImageBackgroundColor },
+                    ReadOnly = new { Changed = readOnlyChanged, Value = paramReadOnly },
+                } ) );
+            }
+        }
+
+        await base.SetParametersAsync( parameters );
+    }
+
+    /// <inheritdoc/>
+    protected override async Task OnAfterRenderAsync( bool firstRender )
+    {
+        if ( firstRender )
+        {
+            DotNetObjectRef ??= DotNetObjectReference.Create( this );
+
+            JSModule = new JSSignaturePadModule( JSRuntime, VersionProvider );
```

Comment:
Most, if not all, of extensions, are done like this. If would to add a DI support, it would mean that we also have to have a register method at `Startup.cs`, eg. `.AddSignaturePad()`. Unless we really need it, I tend to avoid it.